### DISCUSSION TITLE
refetch used to be undefined if enabled: false prior to wagmi 1.x.x version

### DISCUSSION THREAD

**Discussion by Yuripetusko** [wevm/wagmi#2995] [Q&A]
As title says, prior to wagmi 1.x.x / viem, I am almost sure that refetch function on useContractRead was undefined if passing enabled: false to the hook? And now it always seems to be defined. Which makes it harder to conditionally call refetch as there's no context of `enabled` value if you are passing refetch as prop to other components.

**Comment by tmm** [comment]
It should always be defined. Follows the [same behavior as described here](https://tanstack.com/query/v4/docs/react/reference/useQuery).
